ddtrace/internal/telemetry/writer.py

```diff
@@ -382,10 +389,19 @@ def add_integration(self, integration_name, auto_enabled):
             "enabled": True,
             "auto_enabled": auto_enabled,
             "compatible": True,
-            "error": "",
+            "error": self._error[1],  # the integration error only takes a message, no code
         }
+        # Reset the error after it has been reported.
+        self._error = (0, "")
         self._integrations_queue.append(integration)
 
+    def add_error(self, code, msg):
```

Comment:
what happens if we set an error after the app started event is sent?

Reply:
It gets stored and never sent. We will need to implement the `app-configurations-changed` event to send errors after startup. This is out of scope for now.